src/org/exist/indexing/IndexController.java

```diff
@@ -179,6 +175,10 @@ public void setDocument(final DocumentImpl doc, final ReindexMode mode) {
         setMode(mode);
     }
 
+    public void setURL(XmldbURI uri) {
```

Comment:
What is the purpose of this new method? It appears only to be used for `streamMetas`... if so, why not pass it as an argument to `streamMetas`?